src/main/java/org/wso2/carbon/securevault/hashicorp/repository/HashiCorpSecretRepository.java

```diff
@@ -98,12 +110,23 @@ public void init(Properties properties, String id) {
 
             String version = hashiCorpVaultConfigLoader.getProperty(ENGINE_VERSION_PARAMETER);
             engineVersion = !StringUtils.isEmpty(version) ? Integer.parseInt(version) : DEFAULT_ENGINE_VERSION;
-            retrieveRootToken();
+
+            String authTypeConfig = hashiCorpVaultConfigLoader.getProperty(AUTH_TYPE);
+            authType = AuthType.valueOf(authTypeConfig);
+
+            if (authType.equals(AuthType.APP_ROLE)) {
+                roleId = hashiCorpVaultConfigLoader.getProperty(ROLE_ID_PARAMETER);
+                retrieveSecreteId();
```

Comment:
Return the secret id value and assign it from here for better readability.

Reply:
Corrected in the refactor commit 913abdc33b35c37b20e73ed5189933bf23648a05